src/Controller/OpenIDConnectController.php

```diff
@@ -102,11 +102,8 @@ class OpenIDConnectController implements ContainerInjectionInterface {
    * Constructor.
    */
   public function __construct(AuthProviderService $authProviderService, RequestStack $requestStack, SessionInterface $session, CacheItemPoolInterface $cacheItemPool, LanguageManagerInterface $languageManager, LoggerInterface $logger, RendererInterface $renderer) {
-    $plugin = $authProviderService->getActivePlugin();
-    if (!$plugin instanceof OpenIDConnect) {
-      throw new AuthenticationException(sprintf('Invalid plugin: %s; Expected %s', get_class($plugin), OpenIDConnect::class));
-    }
-    $this->plugin = $plugin;
+    $reflect = new \ReflectionClass(OpenIDConnect::class);
+    $this->plugin = $authProviderService->getPluginInstance($reflect->getShortName());
```

Comment:
Unless I'm missing something we should use 

```php
    $this->plugin = $authProviderService->getPluginInstance('OpenIDConnect');
```

here to make sure that we get the right plugin (defined in https://github.com/itk-dev/os2forms_nemlogin_openid_connect/blob/main/src/Plugin/os2web/NemloginAuthProvider/OpenIDConnect.php#L22-L30).

In most case the plugin id (defined in `@AuthProvider`) will match the class name, but we should not rely on that and always use the plugin id to get a plugin instance.

Reply:
idea was to avoid using hardcoded text, in this case, the plugin ID. As it was matching the class name, i decided to use that instead of a hardcoded string. 
Ideally we would have to read it from the annotation (as `class OpenIDConnect -> @AuthProvider -> id`), but i haven't really found any good way of doing that. Seems to be very long path, which does not pay off.

I changed it to simple string now.